code/modules/projectiles/guns/projectile/CMturret.dm

```diff
@@ -0,0 +1,223 @@
+/mob/living/carbon/human
+	var/obj/structure/machinegun/mounted
+
+/mob/living/carbon/human/ClickOn(atom/A, params)
+    if(mounted && Adjacent(mounted))
+        if(next_move < world.time)
+            mounted.shoot(get_turf(A))
+    else
+        . = ..()
+
+
+/obj/structure/machinegun
+	name = "machine gun"
+	desc = "A stationary machine gun."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "mgun+barrier"
+	var/fire_sound = 'sound/weapons/Gunshot_smg.ogg'
+	var/empty_sound = 'sound/weapons/empty.ogg'
+	var/ammo_type = "/obj/item/projectile/bullet/a762"
+	var/ammo = 500
+	var/ammomax = 500
+	var/list/row1 = list()
+	var/list/row2 = list()
+	var/list/row3 = list()
+	var/mob/living/carbon/human/user
```

Comment:
This probably won't GC well. You need to make sure it boots the user out, and nulls references to it in its `Destroy()` proc. Also that whole `process()` proc makes my eyes bleed, what kind of mad person thought it'd be a good idea to handle the person mounted on the turret in `process()`?
Kind of tired and haven't looked through much else, but I'll add some more comments tomorrow. I'd advise you avoid merging for now @Fox-McCloud.